core/app/finders/spree/products/find.rb

```diff
@@ -152,6 +158,15 @@ def by_option_value_ids(products)
         products.where(id: product_ids)
       end
 
+      def by_properties(products)
+        return products unless properties?
+
+        products.
+          joins(:product_properties).
+          where(spree_product_properties: { value: properties.values }).
```

Comment:
@aplegatt @przemosk we need to filter by a combination of option type id and values ids eg.

`products?options[<option_type_id]=<option_values_ids>`

eg.

`products?options[1]=1,2,3&options[2]=1,2,3`